File: hw/aimacana/u3/POO/java/rpg-system/src/main/java/ec/edu/espe/rpg/repository/CharacterRepository.java
package ec.edu.espe.rpg.repository;

import ec.edu.espe.rpg.model.entities.Character;
import java.util.List;

public interface CharacterRepository {
    void save(Character character);
    Character findById(String id);
    List<Character> findAll();
    void delete(String id);
}

File: hw/aimacana/u3/POO/python/rpg-system/src/repository/character_repository.py
from abc import ABC, abstractmethod
from typing import List, Optional
from models.entities.character import Character

class CharacterRepository(ABC):
    @abstractmethod
    def save(self, character: Character):
        pass

    @abstractmethod
    def find_by_id(self, char_id: str) -> Optional[Character]:
        pass

    @abstractmethod
    def find_all(self) -> List[Character]:
        pass

    @abstractmethod
    def delete(self, char_id: str):
        pass
